Add Financial Phrasebank scenario

yifanmai opened the pull request:
A sentiment classification benchmark based on the dataset from Good Debt or Bad Debt - Detecting Semantic Orientations in Economic Texts [(Malo et al., 2013)](https://arxiv.org/abs/1307.5336).

yifanmai commented:
Hi @ryokawajp and @mtake, please review this pull request.

By approving, you acknowledge that:

- You will be listed as a co-author of the merge commit
- You agree to license your work under Apache 2.0

ryokawajp reviewed (commented):
I did a quick test with a local GPT2 and found that most of the outputs were empty. A few outputs contain messages like `The long-term contract` instead of the labels.
```
entries: [
  {description: "financial_phrasebank:model=openai/gpt2", priority: 1},
 ]
```

I suppose that this is because the implemented code uses a different prompt configuration. Previously, we were using `input_noun=None` and `output_noun="Label"`, which are now changed. 

https://github.com/IBM/helm-enterprise-benchmark/blob/main/src/helm/benchmark/run_specs.py#L2602

In addition, although (in v0.4.0,) we did not specify the number of in-context learning shots explicitly,  `max_train_instances: int = 5` was used since that was the default value of get_generation_adapter_spec(). However, the current version does not use ICL now. I suppose the default value has been changed.

ryokawajp commented:
Let me correct my previous comment. I confirmed that the default number of training examples `max_train_instances: int = 5` is kept the same in `get_generation_adapter()`. The actual number can be affected by the length of the texts in the samples as well as by the input context length limit. 
Thus, the only difference from our initial version is that the current version uses different input / output nouns.

yifanmai commented:
Hi @ryokawajp, I made the following fixes:

- Added the in-context learning examples
- Changed the output noun to "Label"
- Changed the input noun to "Sentence"

Please take another look.

ryokawajp reviewed (approved):
LGTM.
Thank you for the update!
